libgadget/density.c

```diff
@@ -88,6 +88,11 @@ typedef struct {
 struct DensityPriv {
     MyFloat *Left, *Right;
     MyFloat (*Rot)[3];
+    /* This is the DhsmlDensityFactor for the pure density,
+     * not the entropy weighted density. It is used here and
+     * copied to DhsmlEgyDensityFactor if DensityIndependentSphOn = 1
+     * If DensityIndependentSphOn = 0 it is used to set DhsmlEgyDensityFactor. */
+    MyFloat * DhsmlDensityFactor;
```

Comment:
This comment is a bit confusing. If DensityIndependentSphOn is false there is no DhsmlEgyDensityFactor in the force equations. Probably you mean DhsmlEgyDensityFactor and DhslmDensityFactor are the same if not doing densityindependentSph?

Reply:
Thanks, clarified.